src/boot/efi/boot.c

```diff
@@ -41,6 +41,11 @@ typedef struct {
         EFI_STATUS (*call)(VOID);
         BOOLEAN no_autoselect;
         BOOLEAN non_unique;
+        UINTN tries_done;
```

Comment:
well, but that would suggest the counter could actually go negative, but it can't... We use (UINTN) -1 as a special way to indicate one particular exceptional case (and we do this similar elsewhere in the tree), hence I don't think we should really change the type altogether....